src/main/java/edu/cornell/kfs/fp/document/validation/impl/CuDisbursementVoucherDocumentPreRules.java

```diff
@@ -67,11 +69,30 @@ public boolean doPrompts(Document document) {
 
         // Handle custom pre-rules:
         preRulesOK &= this.checkCheckStubTextOverflow((DisbursementVoucherDocument) document);
+        
+        setIncomeClassNonReportableForForeignVendorWithNoTaxReviewRequired(document);
 
         return preRulesOK;
     }
 
-    /**
+	private void setIncomeClassNonReportableForForeignVendorWithNoTaxReviewRequired(Document document) {
+		DisbursementVoucherDocument dvDoc = (DisbursementVoucherDocument) document;
+		DisbursementVoucherPayeeDetail dvPayeeDetail = dvDoc.getDvPayeeDetail();
+		
+		String payeeTypeCode = dvPayeeDetail.getDisbursementVoucherPayeeTypeCode();
+		String paymentReasonCode = dvPayeeDetail.getDisbVchrPaymentReasonCode();
+		Integer vendorHeaderId = dvPayeeDetail.getDisbVchrVendorHeaderIdNumberAsInteger();
+
+		if (getCuDisbursementVoucherTaxService().isForeignVendorAndTaxReviewNotRequired(payeeTypeCode,paymentReasonCode, vendorHeaderId)) {
+			dvDoc.getDvNonResidentAlienTax().setIncomeClassCode(DisbursementVoucherConstants.NRA_TAX_INCOME_CLASS_NON_REPORTABLE);
```

Comment:
I'm wondering whether the PreRules is the correct place to be setting this value, since there doesn't appear to be any routing stage or user prompt restrictions to block the update. It would be good if someone else could chime in on whether they think such updates should be moved to a different part of the code if the restrictions mentioned earlier do not apply.